### ISSUE TITLE
Enforce consistent attribute ordering

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#387]
From @AliSoftware:

> Like enforce to consistently use `@IBOutlet private weak var` and not `weak private @IBOutlet var` for example.

Since these are always language keywords, it should be super easy to write a rule that does this. The hard part would be settling on an ordering convention.

**Comment by AliSoftware** [comment]
Du you have an exhaustive list of possible keywords (possibly extracted from SourceKitten) so we could discuss the ordering?

I see `@IBOutlet`/`@IBInspectable` `weak`/`unowned`, `lazy`,`public`/`private`/`internal`,`@objc`… probably more?

**Comment by jpsim** [comment]
I'm sure there are many more. Maybe there's an exhaustive list over at [/apple/swift](/apple/swift)?

**Comment by ikesyo** [comment]
- Modifires: https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Declarations.html#//apple_ref/swift/grammar/declaration-modifiers
- Attributes: https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Attributes.html#//apple_ref/doc/uid/TP40014097-CH35-ID347

**Comment by jpsim** [comment]
Thanks!

**Comment by masters3d** [comment]
As far as I can tell, attributes are enforced by the compiler to be declared first. 
These will not compile.
```swift
final public @objc class MyCoolClass: NSObject {} // expected declaration
final @objc public class MyCoolClass: NSObject {} // expected declaration
public @objc final class SomethingCool:NSObject {} // expected declaration
```
This only leaves us with Access Control Labels and other Modifiers to order:
I have a draft implementation for this:
`<@Attibutes> <ACL> <OtherModifiers>`

**Comment by masters3d** [comment]
I've hit a little snag here. All of these modifiers are `source.lang.swift.syntaxtype.attribute.builtin` except:
`class` and `final`. 

`final` is actually`source.lang.swift.syntaxtype.keyword`

```
GRAMMAR OF A DECLARATION MODIFIER

declaration-modifier → class­  convenience­  dynamic­  final­  infix­  lazy­ optional­  override­  postfix­  prefix­  required­  static­  unowned­  unowned­(­safe­)­  unowned­(­unsafe­)­  weak­
declaration-modifier → access-level-modifier­
declaration-modifier → mutation-modifier­
declaration-modifiers → declaration-modifier­declaration-modifiers­opt­
access-level-modifier → private­  private­(­set­)­
access-level-modifier → fileprivate­  fileprivate­(­set­)­
access-level-modifier → internal­  internal­(­set­)­
access-level-modifier → public­  public­(­set­)­
access-level-modifier → open­  open­(­set­)­
mutation-modifier → mutating­  nonmutating­
```

**Comment by mildm8nnered** [comment]
Isn't this resolved by `modifier_order` #1501?

**Comment by SimplyDanny** [comment]
> Isn't this resolved by `modifier_order` #1501?

I think so. Let's close this.
